interfaces/opensearch_client/v0/README.md

```diff
@@ -0,0 +1,82 @@
+# `opensearch_client`
+
+## Usage
+
+This relation interface describes the expected behaviour of any charm interfacing with the [Charmed Opensearch Operator](https://github.com/canonical/opensearch-operator) using the `opensearch-client` relation.
```

Comment:
We may not want to make this specific to the `opensearch-operator`. For example one could use this same interface with an elasticsearch operator

Reply:
Yeh, this is an interesting one! Are we using any features that are Opensearch specific, or are we mainly using Opensearch in a way that is compliant with the Elasticsearch API? Should this be elasticsearch_client? @delgod / @deusebio ?

Reply:
Yes, I agree. I'd say it would be best to make it general and allow this to be applied to both elasticsearch and opensearch. Actually, I'm actually wondering whether it would be good to make it potentially extensible for zinc and solr too. Maybe calling the interface in a bit more general way, like "indexing_client". Then, in the `version` field (provided back by the Provider) the requirer can specify the details on the specific product (elasticsearch, opensearch, zinc, solr, etc) as well as the version.

Reply:
Yeh, I was planning on robbing this for Zinc when it lands, but the key is that Zinc presents an Elasticsearch compatible API

Reply:
If we plan to also support other products beside elasticsearch, maybe I'd use a name which is a bit more general. We can use the version field to provide details about the underlying technology and it is up to the requirer side to decide which api he wants to use/make sure it is compatible

This sounds to me an approach which is similar to what we are planning and discussed for s3, where we have one interface for conceptually similar products/technologies, using another field to provide extra info about which implementation the provider has

Reply:
how about `fulltext_search` then?

Reply:
Perhaps the answer here is just to KISS and leave this as `opensearch_client` for now? 

I feel like `fulltext_search` is a little ambiguous, given that all of the implementations have some sort of specificity that would be lost in that name. 

I think in the description we should just note that the interface is very likely to satisfy the requirement for any Elasticsearch API-compatible application.

Reply:
Yes, I really don't have very strong opinion, and I'm very fine with the original name (which is also possibly more inline with the as-is, see below). But I believe it is fair to have a thought about this as this seems to often lead to discussions. 

I can see the pattern that we want interfaces to be more general than a specific implementation/product (e.g. S3 and OpenSearch, SQL). The detail of the specific implementation should be provided in a separated field and it is up to the requirer side charm author to handle this correctly (throw an exception if the specific implementation is not supported). The name of the interface is the one that led to the first implementation (or we could say the opensource name could be the rational, hence opensearch here), and if new implementation partially share the same API, we should fallback to the already existing one (with the open source name). This would make the choice of "opensearch" also more consistent with what we have now elsewhere, e.g. "mysql_client"

Reply:
Can I take this as consensus that `opensearch_client` is acceptable, at least for now? I've updated the description to ask for contributions if it's not usable for other elasticsearch-compatible applications.

Reply:
Yep! Thanks!